nrf-hal-common/src/twim.rs

```diff
@@ -132,37 +122,116 @@ where
             // values.
             unsafe { w.maxcnt().bits(buffer.len() as _) });
 
-        // Clear address NACK.
-        self.0.errorsrc.write(|w| w.anack().bit(true));
+        Ok(())
+    }
 
-        // Start write operation.
-        self.0.tasks_starttx.write(|w|
-            // `1` is a valid value to write to task registers.
-            unsafe { w.bits(1) });
+    /// Set RX buffer, checking that it has suitable length.
+    fn set_rx_buffer(&mut self, buffer: &mut [u8]) -> Result<(), Error> {
```

Comment:
Since the lifetime constraints aren't enforced, this should probably be `unsafe` (even if it's private)
```suggestion
    unsafe fn set_rx_buffer(&mut self, buffer: &mut [u8]) -> Result<(), Error> {
```